gson/src/main/java/com/google/gson/JsonPrimitive.java

```diff
@@ -260,6 +264,11 @@ public int hashCode() {
     return value.hashCode();
   }
 
+  /**
+   * Returns whether the other object is equal to this. This method only considers
+   * the other object to be equal if it is an instance of {@code JsonPrimitive} and
+   * has an equal value.
```

Comment:
I did not want to describe in detail how number comparison is performed because there are multiple open issues which show that there are potential flaws with the current logic.